### ISSUE TITLE
add `linux-headers/arch/arm64` since we only have `x86`

### ISSUE THREAD

**Issue by vogler** [goblint/analyzer#312]
On Apple M1 many regression tests fail due to the wrong `x86` headers:
```
goblint/analyzer/linux-headers/arch/x86/include/asm/pvclock.h:56:2: error: #error implement me!
  -    56 | #error implement me!
  -       |  ^~~~~
  - Goblint: Preprocessing failed.
```

`pvclock.h`:
```c
#ifdef __i386__
	__asm__ (
		"mul  %5       ; "
		"mov  %4,%%eax ; "
		"mov  %%edx,%4 ; "
		"mul  %5       ; "
		"xor  %5,%5    ; "
		"add  %4,%%eax ; "
		"adc  %5,%%edx ; "
		: "=A" (product), "=r" (tmp1), "=r" (tmp2)
		: "a" ((u32)delta), "1" ((u32)(delta >> 32)), "2" (mul_frac) );
#elif defined(__x86_64__)
	__asm__ (
		"mulq %[mul_frac] ; shrd $32, %[hi], %[lo]"
		: [lo]"=a"(product),
		  [hi]"=d"(tmp)
		: "0"(delta),
		  [mul_frac]"rm"((u64)mul_frac));
#else
#error implement me!
#endif
```

I think these headers are pretty old, maybe it's also a good idea to update them.

**Comment by michael-schwarz** [comment]
> maybe it's also a good idea to update them.

If CIL can parse the newer ones, absolutely!

**Comment by vogler** [comment]
@vesalvojdani I tried to use a newer kernel, but config fails on my machine:

~~~console
$ mv linux-headers linux-headers-old
$ wget https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.14.2.tar.xz
$ tar xf linux-5.14.2.tar.xz
$ mv linux-5.14.2 linux-headers
$ make test
Testing 00/14 sanity/startstate	 Status: 1 (error)
  - /Users/voglerr/goblint/analyzer/linux-headers/include/linux/kconfig.h:5:10: fatal error: generated/autoconf.h: No such file or directory
  -     5 | #include <generated/autoconf.h>
  -       |          ^~~~~~~~~~~~~~~~~~~~~~
  - compilation terminated.
  - Goblint: Preprocessing failed.
$ cd linux-headers
$ make
***
*** Configuration file ".config" not found!
***
*** Please run some configurator (e.g. "make oldconfig" or
*** "make menuconfig" or "make xconfig").
***
$ make menuconfig
gcc -no-integrated-as --prefix= -Werror=unknown-warning-option: unknown assembler invoked
scripts/Kconfig.include:50: Sorry, this assembler is not supported.
make[2]: *** [menuconfig] Error 1
make[1]: *** [menuconfig] Error 2
make: *** [__sub-make] Error 2
$ cp ../linux-headers-old/config.txt .config
$ make oldconfig
gcc -no-integrated-as --prefix= -Werror=unknown-warning-option: unknown assembler invoked
scripts/Kconfig.include:50: Sorry, this assembler is not supported.
~~~

**Comment by sim642** [comment]
I think @vesalvojdani tried something, but then found out that newer kernels use some C11 features like `_Generic`, which CIL (and Goblint) don't currently support (https://github.com/goblint/cil/issues/13, https://github.com/goblint/cil/pull/24).

**Comment by vogler** [comment]
We can also use an older version (our `config.txt` says `Linux/x86 4.0.0`), but idk if the introduction of those C11 features happened before `arch/arm64` was added.

**Comment by vogler** [comment]
Also tried `make headers_install ARCH=arm64 INSTALL_HDR_PATH=usr` ([doc](https://www.kernel.org/doc/html/latest/kbuild/headers_install.html)) which needs no config, but that's not enough: `analyzer/linux-headers/include/linux/kconfig.h: No such file or directory`.

**Comment by vogler** [comment]
https://github.com/goblint/analyzer/blob/c3932dce28f397dbbf7b0f8f98842de2950a08dd/src/maingoblint.ml#L171
This (and #54) would then be determined by an option `arch` defaulting to the current machine's arch.
